cli/azd/internal/appdetect/docker.go

```diff
@@ -1,19 +1,67 @@
 package appdetect
 
 import (
+	"bufio"
 	"io/fs"
+	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
 )
 
 func detectDocker(path string, entries []fs.DirEntry) (*Docker, error) {
 	for _, entry := range entries {
 		if strings.ToLower(entry.Name()) == "dockerfile" {
+			dockerFilePath := filepath.Join(path, entry.Name())
+			file, err := os.Open(dockerFilePath)
+			if err != nil {
+				return nil, err
+			}
+			defer file.Close()
+			scanner := bufio.NewScanner(file)
+
+			var ports []Port
+			for scanner.Scan() {
+				line := scanner.Text()
+				if strings.HasPrefix(line, "EXPOSE") {
+					parsedPorts, err := parsePorts(line[len("EXPOSE"):])
+					if err != nil {
+						return nil, err
```

Comment:
Looking at this once more, I now realize that your original implementation of not failing when we fail to parse `EXPOSE` is much better. We end up falling back to prompting the user.

My suggestion would be to log the error and move on:
```suggestion
				log.Errorf("parsing Dockerfile at %s: %v", dockerFilePath, err)
```

Reply:
Updated.